### PR TITLE
set relative field layout based on alliance from FMS

### PR DESCRIPTION
**Pull request by northeastprince**

### CODE REVIEW

**Comment by prensing** [comment]
Matt, this looks sensible. However, there is a major failure mode. The code you wrote sets the field orientation on robot initialization, but the robot won't connect to the FMS for many seconds. Even in our shed, the robot boots before the Driver's Station connects.  At a competition, it can actually take minutes to get everything connected.

So, to start, I think the Vision needs a method to change the field orientation. Then, we need to figure out where in the Robot methods to make that call, and how to know that it is the "final" answer.

**Comment by prensing** [comment]
Matt, I think this will work, so we should test it when we get a chance. I would like to try to refine it. If we just change the origin at autonomousInit(), the robot will have been sitting waiting for the match to start and accumulating position readings for the wrong position (because it is seeing an AprilTag all that time). Then it will try to start it's auto while all of a sudden it position is wrong.
We should ask on ChiefDelphi, but I assume the best thing to do is: 
* include tests for FMS and driver station connects (these are in the DriverStation class)
* run this while disabled (disabledPeriodic).

**Comment by We1chJ** [comment]
Looks good to me

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -132,4 +143,14 @@ public void testInit() {
     @Override
     public void testPeriodic() {
     }
+
+    private void checkDSUpdate() {
```
Comment (prensing) [inline src/main/java/frc/robot/Robot.java:147]: This routine needs comments to say why it is called and what it does.

**Review by prensing — CHANGES_REQUESTED** [review]
Matt, the code looks fine, but you need to comment everything. The routines should have a comment, and you definitely should comment *why* we are calling checkDSUpdate in 2 places.

File: src/main/java/frc/robot/commands/FollowTrajectory.java [hunk 2]
```diff
@@ -90,9 +91,13 @@ public FollowTrajectory(DriveTrain driveTrain,
 	// Called when the command is initially scheduled.
 	@Override
 	public void initialize() {
+        if (m_actualTrajectory == null) {
+            m_actualTrajectory = PathPlannerTrajectory.transformTrajectoryForAlliance(m_trajectory, DriverStation.getAlliance());
```
Comment (prensing) [inline src/main/java/frc/robot/commands/FollowTrajectory.java:95]: Unfortunately, this is going to cause problems. If this is an Auto trajectory, we presumably do want to transform the path. However, we are also using this for on-the-fly trajectories, where we don't want the transform. I guess we should add a boolean to this command?

**Pull request closed without merging**